ambry-metrics/src/main/java/com.github.ambry.metrics/JvmMetrics.java

```diff
@@ -80,24 +80,34 @@ public JvmMetrics(MetricsRegistry registry) {
   }
 
   public void start() {
-    executor.startup();
-    executor.schedule("jvmmetrics", this, 0, 5, TimeUnit.SECONDS);
+    executor.scheduleAtFixedRate(this, 0, 5, TimeUnit.SECONDS);
   }
 
   public void run() {
-    logger.debug("updating jvm metrics");
-
-    updateMemoryUsage();
-    updateGcUsage();
-    updateThreadUsage();
-
-    logger.debug("updated metrics to: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}]", gMemNonHeapUsedM,
-        gMemNonHeapCommittedM, gMemHeapUsedM, gMemHeapCommittedM, gThreadsNew, gThreadsRunnable, gThreadsBlocked,
-        gThreadsWaiting, gThreadsTimedWaiting, gThreadsTerminated, cGcCount, cGcTimeMillis);
+    try {
+      logger.debug("updating jvm metrics");
+
+      updateMemoryUsage();
+      updateGcUsage();
+      updateThreadUsage();
+
+      logger.debug("updated metrics to: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}]", gMemNonHeapUsedM,
+          gMemNonHeapCommittedM, gMemHeapUsedM, gMemHeapCommittedM, gThreadsNew, gThreadsRunnable, gThreadsBlocked,
+          gThreadsWaiting, gThreadsTimedWaiting, gThreadsTerminated, cGcCount, cGcTimeMillis);
+    } catch (Exception e) {
+      logger.error("Encountered exception while updating jvm metrics.", e);
+    }
   }
 
   public void stop() {
-    executor.shutdown();
+    try {
+      executor.shutdown();
+      if (!executor.awaitTermination(1, TimeUnit.DAYS)) {
```

Comment:
this was the default for the scheduler class before, so I just kept it for production code pieces